torch/_dynamo/variables/torch.py

```diff
@@ -836,10 +836,11 @@ def speculate_subgraph(f, sub_args, graph_checkpoint, checkpoint):
             # Setup the subgraph we're going to capture into
             tx.output.graph = torch.fx.Graph()
             tx.output.graphargs = []
-            tx.output.name_to_input.clear()
+            tx.output.input_name_to_proxy.clear()
 
             args = []
             # One argument to graph per sub_args
+            # TODO (awgu): Do we need to de-dup here?
```

Comment:
This comment is not meant to be landed. I need to figure out if we need to change the logic here.

Reply:
These arguments all represent locals, so no, I don't think they need to be deduped. The relevant case is

```
torch.cond(pred, true_fn, false_fn, [x, x])
```

you could try to test it.